src/renderer/contextMenu/editor/index.js

```diff
@@ -15,15 +15,36 @@ import {
   INSERT_COLUMN,
   REMOVE_COLUMN
 } from './menuItems'
+import spellcheckMenuBuilder from './spellcheck'
 
 const { Menu, MenuItem } = remote
 
-export const showContextMenu = (event, { start, end }) => {
+/**
+ * Show editor context menu.
+ *
+ * @param {MouseEvent} event The native mouse event.
+ * @param {*} selection The editor line with start and end offset.
+ * @param {[SpellChecker]} spellchecker The spellcheck wrapper.
+ * @param {[string]} selectedWord The selected word.
+ * @param {[string[]]} wordSuggestions Suggestions for `word`.
+ * @param {*} replaceCallback The callback to replace the word by a replacement.
+ */
+export const showContextMenu = (event, selection, spellchecker, selectedWord, wordSuggestions, replaceCallback) => {
+  const { start, end } = selection
   const menu = new Menu()
   const win = remote.getCurrentWindow()
   const disableCutAndCopy = start.key === end.key && start.offset === end.offset
   const CONTEXT_ITEMS = [INSERT_BEFORE, INSERT_AFTER, SEPARATOR, CUT, COPY, PASTE, SEPARATOR, COPY_AS_MARKDOWN, COPY_AS_HTML, PASTE_AS_PLAIN_TEXT]
 
+  const spellingSubmenu = spellcheckMenuBuilder(spellchecker, selectedWord, wordSuggestions, replaceCallback)
+  if (spellingSubmenu) {
+    menu.append(new MenuItem({
+      label: 'Spelling...',
+      submenu: spellingSubmenu
```

Comment:
I prefer put `wordSuggestions` at the top most content menu item, not in the submenu. like bellow:
<img width="657" alt="屏幕快照 2019-10-12 下午4 34 06" src="https://user-images.githubusercontent.com/9712830/66698265-f5c27e00-ed0e-11e9-901c-b5576c2301a2.png">

BTW, what's the function of `change language...` in context menu, does it change the setting in preference?

Reply:
> I prefer put wordSuggestions at the top most content menu item, not in the submenu. like bellow:

Me too but I don't think it's not a good choice because the menu gets very big (up to 10/15 words can be suggested).

> BTW, what's the function of change language... in context menu, does it change the setting in preference?

It's change the spellchecker language without changing the settings value.